### ISSUE TITLE
Peers are not reconnected when ICE fails in a non "stable" signaling state

### ISSUE THREAD

**Issue by danxuliu** [nextcloud/spreed#1630]
After an ICE failure an ICE restart may be triggered. However, if there is another ICE failure while the negotiation is still happening then no further ICE restarts are triggered.

The reason is that [ICE restarts are only triggered when the signaling state is `stable`](https://github.com/nextcloud/spreed/blob/1d0b7f9d433004d1891eaf47c790e9f2623568b4/js/webrtc.js#L524), but if the error happened for example after creating the offer the signaling state would be `have-local-offer` instead.

This is unrelated to the [new SimpleWebRTC bundle](https://github.com/nextcloud/spreed/pull/1524).

**Comment by nickvergessen** [comment]
Still an issue?

**Comment by tdm4** [comment]
Hi @nickvergessen - Not an issue now. I just upgraded to Nextcloud 18.0.0 so I'm going to try another 3-way talk tomorrow, but the past few weeks I've used it, it's not had an issue. Sometimes it takes a little while to set up when the 3rd person joins, but I think there's some network routing packet drops between one of the parties and the nextcloud talk server, so that could be why it takes a little longer to set up.

**Comment by danxuliu** [comment]
Yes, [it is still an issue](https://github.com/nextcloud/spreed/blob/8288d49c7edaf3d6d242712e35f036b64e18ec7f/src/utils/webrtc/webrtc.js#L514).

**Comment by tdm4** [comment]
@danir-de I think it's some issue with signalling in the internal nextcloud php code.
